### PR TITLE
feat(build): Add containerd v1.4.0 CRI CI test

### PR DESCRIPTION
**Pull request by hickeyma**
Partial #613

### CODE REVIEW

**Comment by hickeyma** [comment]
Draft until add step to retrieve the containerd cri tag/sha which corresponds to the containerd version

File: 1 [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Add containerd v1.4.0 CI test
```
Comment (mikebrow) [inline 1:1]: i believe this file is an oops...

Comment (hickeyma) [reply 1]: Yep, a randomer! 😁

**Comment by hickeyma** [comment]
> nits on the draft so far:
let's consider splitting the containerd and crio tests into separate ci yamls
say containerd.yml, crio.yml, and later on docker.yml
otherwise you guys will be rebasing a lot

Yep, agree. It was on my list todo.

**Comment by hickeyma** [comment]
> Draft until add step to retrieve the containerd cri tag/sha which corresponds to the containerd version

Done ✔️

**Comment by hickeyma** [comment]
> let's consider splitting the containerd and crio tests into separate ci yamls

Waiting on #622 to merge then will split out and open PR for review.

**Comment by feiskyer** [comment]
#622 merged now.

**Comment by hickeyma** [comment]
Thanks for the heads up @feiskyer

**Comment by hickeyma** [comment]
> let's consider splitting the containerd and crio tests into separate ci yamls
say containerd.yml, crio.yml, and later on docker.yml

done ✔️

File: .github/workflows/crio.yml [hunk 2]
```diff
@@ -93,24 +26,20 @@ jobs:
         run: |
           echo "::set-env name=GOPATH::${{ github.workspace }}"
           echo "::add-path::${{ github.workspace }}/bin"
-
       - name: Setup system
         run: |
           # enable necessary kernel modules
           sudo ip6tables --list >/dev/null
-
```
Comment (saschagrunert) [inline .github/workflows/crio.yml:101]: I personally prefer to keep some optical separation (not blocking).

Comment (hickeyma) [reply .github/workflows/crio.yml:101]: Totally agree. Something seemed to get messed up in splitting the files. It is addressed now.

File: .github/workflows/containerd.yml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+jobs:
+  #
+  # Run CRI tests against containerd
+  #
+  linux-build-and-test-containerd:
+    strategy:
+      matrix:
+        version: [master, v1.4.0-beta.1]
+    name: critest - containerd ${{matrix.version}} (linux amd64)
```
Comment (mikebrow) [inline .github/workflows/containerd.yml:12]: a little wordy to say Containerd CI/critest -containerd/

Comment (mikebrow) [reply .github/workflows/containerd.yml:12]: suggest "containerd ${{matrix.version}} arch"

File: .github/workflows/containerd.yml [hunk 4]
```diff
@@ -0,0 +1,91 @@
+name: Containerd CI
```
Comment (mikebrow) [inline .github/workflows/containerd.yml:1]: suggest name critest

File: .github/workflows/containerd.yml (hunk 3 above)
Comment (mikebrow) [reply .github/workflows/containerd.yml:12]: because of the matrix options this will get pretty long..

Comment (mikebrow) [reply .github/workflows/containerd.yml:12]: or you could make the top name just containerd.. and mention critest for this name.. either way works..

File: .github/workflows/crio.yml [hunk 5]
```diff
@@ -171,9 +99,9 @@ jobs:
...
             --parallel=$(nproc)
           sudo journalctl -u crio > cri-o.log
-
+          
       - name: Upload CRI-O logs
         uses: actions/upload-artifact@v1
         with:
           name: cri-o-${{matrix.version}}-${{github.sha}}.log
-          path: cri-o.log
+          path: cri-o.log
```
Comment (mikebrow) [inline .github/workflows/crio.yml:107]: missing new line at end of file

File: .github/workflows/containerd.yml [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+      - name: CRI Test on this cri-tools commit
+        run: |
+          make test-cri
+        working-directory: ${{ github.workspace }}/src/github.com/containerd/cri
+
+      - name: Upload containerd ${{matrix.version}} logs
+        uses: actions/upload-artifact@v1
+        with:
+          name: containerd-${{matrix.version}}-${{github.sha}}.log
+          path: /tmp/test-cri/containerd.log
```
Comment (mikebrow) [inline .github/workflows/containerd.yml:91]: missing newline at end of file

File: .github/workflows/containerd.yml (hunk 3 above)
Comment (hickeyma) [reply .github/workflows/containerd.yml]: Ok, changed it. Lets see what you think.

File: .github/workflows/containerd.yml (hunk 4 above)
Comment (hickeyma) [reply .github/workflows/containerd.yml]: Ok, changed it. Lets see what you think.

File: .github/workflows/crio.yml (hunk 5 above)
Comment (hickeyma) [reply .github/workflows/crio.yml]: added

File: .github/workflows/containerd.yml (hunk 6 above)
Comment (hickeyma) [reply .github/workflows/containerd.yml]: added

**Comment by hickeyma** [comment]
Thanks for the review @saschagrunert and @mikebrow. Pushed an update for the review comments.

File: .github/workflows/crio.yml [hunk 7]
```diff
@@ -171,7 +99,7 @@ jobs:
             --ginkgo.flakeAttempts=3 \
             --parallel=$(nproc)
           sudo journalctl -u crio > cri-o.log
-
+
```
Comment (saschagrunert) [inline .github/workflows/crio.yml:102]:
```suggestion

```

File: .github/workflows/containerd.yml [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+
+jobs:
+  #
+  # Run CRI tests against containerd
+  #
+  linux-build-and-test-containerd:
+    strategy:
+      matrix:
+        version: [master, v1.4.0-beta.1]
+    name: containerd ${{matrix.version}} (linux amd64)
```
Comment (saschagrunert) [inline .github/workflows/containerd.yml:12]: What do you like more:
```
critest / containerd master (linux amd64) (pull request)
```
or
```
critest / containerd master / linux amd64 (pull request)
```

File: .github/workflows/crio.yml (hunk 7 above)
Comment (hickeyma) [reply .github/workflows/crio.yml]: fixed

File: .github/workflows/containerd.yml (hunk 8 above)
Comment (hickeyma) [reply .github/workflows/containerd.yml]: Good pint @saschagrunert. I think the second version is more symmetrical. Updated to mirror that.

**Comment by hickeyma** [comment]
Thanks for review @mikebrow and @saschagrunert. Pushed some small changes following comments.

**Comment by saschagrunert** [comment]
Thank you @hickeyma :pray: !

**Comment by saschagrunert** [comment]
Totally strange to me that it initially worked and now got broken. Working on a fix in #626

**Comment by hickeyma** [comment]
> Totally strange to me that it initially worked and now got broken

I agree @saschagrunert. It's as if the runners changed and can't handle the network changes. Thanks for working on a fix. 👏

**Comment by hickeyma** [comment]
Rebased and build/tests succeed. Ready for review again.

**Comment by hickeyma** [comment]
Thanks @saschagrunert for merging.

**Pull request merged**